### PR TITLE
Mlx90614

### PR DESCRIPTION
**Pull request by synco**
# What does this implement/fix? 
Implemented the Melexis MLX90614 thermopile sensor.
Required the write+read from the I2C bus without releasing the bus.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
sensor:
  - platform: mlx90614
    target_temperature:
      id: my_object
      name: "Object Temperature"
      accuracy_decimals: 2
    reference_temperature:
      id: my_ambient
      name: "Ambient"
      accuracy_decimals: 2
    update_interval: 10s
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [coming soon] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/i2c/i2c_bus.h [hunk 1]
```diff
@@ -26,8 +26,35 @@ struct WriteBuffer {
...
+  uint8_t device_address;
+  const uint8_t *write_buf;
+  size_t write_len;
+  uint8_t *read_buf;
+  size_t read_len;
+};
+
 class I2CBus {
  public:
+  /// Most I2C (and SMBus) bus transactions follow the process of
```
Comment (OttoWinter) [inline esphome/components/i2c/i2c_bus.h:39]: 1. If changing the i2c base model, please do so in another PR.
2. I feel like this would be better as just a parameter to the `write(v)` methods, similar to the arduino API, i.e. make it send a STOP signal or not.

**Review by OttoWinter — COMMENTED** [review]
Thanks for this integration! the MLX90614 part of it seems ok, but could you take a look at the i2c api comments? Thanks!

**Comment by jesserockz** [comment]
@synco If you would like to revisit this, i2c now has an optional boolean to not send the stop signal.
